linker/shared/src/test/scala/org/scalajs/linker/backend/javascript/PrintersTest.scala

```diff
@@ -159,6 +159,33 @@ class PrintersTest {
     )
   }
 
+  @Test def objectLitAtStartOfStatement_Issue4949(): Unit = {
```

Comment:
IMO here it makes more sense to simply test the expected behavior for individual trees. (somewhat whitebox testing, depending on how you look at it).

We do not need to test the specific issues we encountered on a global level: We have way better ways of testing these (end-to-end).

As it stands now, IMHO the test is hard to understand and provides little value (it's likely to be brittle in fact).

I would simply add the following two test cases:

```scala
@Test def emptyObjectLiteral(): Unit = {
  assertPrintEquals("({})", ObjectConstr(Nil))
}

@Test def objectLiteral(): Unit = {
  assertPrintEquals(
      """
        |({
        |  "foo": 1
        |})
       """,
       ObjectConstr(List(StringLiteral("foo") -> IntLiteral(1)))
  )
}
```

We might even go as far and say the `emptyObjectLiteral` test is not really useful. It is obvious that the code is doing this so why test it (that what the code is doing works on a JS level is tested elsewhere).

Further, we might say that the `objectLiteral` test should go into a different commit before: The addition / removal of the parenthesis does not really warrant unit testing IMO, but the overall object literal code does (it is non-obvious what it does and a test can help).

Does that make sense?

Reply:
Yes, OK, that makes sense. Updated accordingly.